### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by Pairarma**
There are many unused variable, that may be not necessary at all.
To keep the basic structure of the functions, the variables are used without real meaning.

### CODE REVIEW

**Comment by yoursunny** [comment]
To prevent the same warnings from reappearing, please update CI scripts so that these warnings become errors.

**Comment by Pairarma** [comment]
The .travis.yml script already runs unit-tests and checks if they are successful. Now the tests also check for unused variables, which causes the script to fail. To fix that all unit-tests need to be updated.

**Comment by Pairarma** [comment]
Fixed remaining unused vairable, so that the CI script checks pass.

**Review by yoursunny — APPROVED** [review]

**Comment by Pairarma** [comment]
I noticed that all warnings were 'unused-parameter'. I now added the check for 'unused-variable' too, to also prevent that from happening.

**Review by yoursunny — APPROVED** [review]

File: security/default-backend/ndn-lite-default-rng-impl.c [hunk 1]
```diff
@@ -8,13 +8,16 @@
...
+#include <stddef.h>
 #include "ndn-lite-default-rng-impl.h"
 #include "../ndn-lite-rng.h"
 #include "../../ndn-error-code.h"
 
 /* always fails and return 0 */
 static int ndn_lite_default_rng(uint8_t *dest, unsigned size)
 {
+    if (dest == NULL || size == 0)
+      return 0;
```
Comment (yoursunny) [inline security/default-backend/ndn-lite-default-rng-impl.c:20]: This condition does make sense.
Just `(void*)` unused parameters.

**Review by yoursunny — CHANGES_REQUESTED** [review]

File: security/default-backend/ndn-lite-default-rng-impl.c (hunk 1 above)
Comment (Pairarma) [reply security/default-backend/ndn-lite-default-rng-impl.c]: Fixed with the latest commit (9c74035)

**Review by yoursunny — APPROVED** [review]

**Comment by tianyuan129** [comment]
> @tianyuan129 can you check Travis-CI integration?
> The latest commit isn't built, and the [requests](https://travis-ci.com/github/named-data-iot/ndn-lite/requests) page has error:
> 
> > Could not authorize build request for named-data-iot/ndn-lite.

Fixed

File: app-support/access-control.c [hunk 2]
```diff
@@ -137,6 +137,7 @@ _ac_timeout()
 int
 _on_ac_notification(const uint8_t* interest, uint32_t interest_size, void* userdata)
 {
+  (void*) userdata;
```
Comment (yoursunny) [inline app-support/access-control.c:140]: I compared this to my NDNph codebase. It seems that you only need `(void)userdata`, not `(void*)userdata`.

Comment (Pairarma) [reply app-support/access-control.c]: Changed with the latest commit (f5c8d29)

**Review by yoursunny — APPROVED** [review]

**Comment by Pairarma** [comment]
Using `(void)` for the unused parameters will cause the error `-Werror=unused-variable` when compiling ndn-lite for RIOT. The latest commit (02c96cc) fixed that by ignoring the warnings for the specific lines using pragmas.

**Pull request merged**